File: scripts/solar/common/FourCC.lua
local ____lualib = require("lualib_bundle")
local __TS__Class = ____lualib.__TS__Class
local __TS__SourceMapTraceBack = ____lualib.__TS__SourceMapTraceBack
__TS__SourceMapTraceBack(debug.getinfo(1).short_src, {["6"] = 7,["7"] = 7,["8"] = 7,["10"] = 7,["11"] = 11,["12"] = 13,["13"] = 14,["14"] = 15,["15"] = 16,["16"] = 11,["17"] = 19,["18"] = 20,["19"] = 21,["20"] = 22,["22"] = 24,["23"] = 19,["24"] = 27,["25"] = 29,["26"] = 30,["27"] = 31,["28"] = 32,["29"] = 27,["30"] = 35,["31"] = 37,["32"] = 38,["33"] = 39,["35"] = 41,["36"] = 35,["37"] = 8,["38"] = 9,["39"] = 7});
local ____exports = {}
____exports.default = __TS__Class()
local FourCC = ____exports.default
FourCC.name = "FourCC"
function FourCC.prototype.____constructor(self)
end
function FourCC._id(self, a)
    local r = (">I4"):pack(a)
    self.ids1[a] = r
    self.ids2[r] = a
    return r
end
function FourCC.id2string(a)
    local str = ____exports.default.ids1[a]
    if str then
        return str
    end
    return ____exports.default:_id(a)
end
function FourCC.__id2(self, a)
    local r = (">I4"):unpack(a)
    self.ids2[a] = r
    self.ids1[r] = a
    return r
end
function FourCC.string2id(a)
    local str = ____exports.default.ids2[a]
    if str then
        return str
    end
    return ____exports.default:__id2(a)
end
FourCC.ids1 = {}
FourCC.ids2 = {}
____exports.default = FourCC
return ____exports

File: src/solar/common/FourCC.ts
/**
 * FourCC
 * FourCC全称Four-Character Codes，代表四字符代码 (four character code), 它是一个32位的标示符
 * 此类可将4字符的字符串转换为整数
 * 以及将整数转换为4个字符的字符串
 */
export default class FourCC {
    static ids1: any = {}
    static ids2: any = {}

    static _id(a: number) {
        // @ts-ignore
        let r = '>I4'.pack(a);
        this.ids1[a] = r
        this.ids2[r] = a
        return r
    }

    static id2string(this: void,a: number) {
        let str = FourCC.ids1[a];
        if (str) {
            return str;
        }
        return FourCC._id(a)
    }

    static __id2(a: string) {
        // @ts-ignore
        let r = ('>I4').unpack(a)
        this.ids2[a] = r
        this.ids1[r] = a
        return r
    }

    static string2id(this: void, a: string) {

        let str = FourCC.ids2[a];
        if (str) {
            return str;
        }
        return FourCC.__id2(a)
    }


}
